routes/api/users.js

```diff
@@ -1,101 +1,170 @@
+            })
+            .catch((err) => console.log(err));
         });
-      }
-    });
+      });
+    }
   });
+});
 
-  router.post("/login", (req, res) => {
-    const { errors, isValid } = validateLoginInput(req.body);
-  
-    if (!isValid) {
+router.post("/login", (req, res) => {
+  const { errors, isValid } = validateLoginInput(req.body);
+
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+
+  const email = req.body.email;
+  const password = req.body.password;
+
+  User.findOne({ email }).then((user) => {
+    if (!user) {
+      errors.email = "This user does not exist";
       return res.status(400).json(errors);
     }
-  
-    const email = req.body.email;
-    const password = req.body.password;
-  
-    User.findOne({ email }).then(user => {
-      if (!user) {
-        errors.email = "This user does not exist";
-        return res.status(400).json(errors);
-      }
-  
-      bcrypt.compare(password, user.password).then(isMatch => {
-        if (isMatch) {
-          const payload = { id: user.id, email: user.email };
-  
-          jwt.sign(payload, keys.secretOrKey, { expiresIn: 3600 }, (err, token) => {
+
+    bcrypt.compare(password, user.password).then((isMatch) => {
+      if (isMatch) {
+        const payload = { id: user.id, email: user.email };
+
+        jwt.sign(
+          payload,
+          keys.secretOrKey,
+          { expiresIn: 3600 },
+          (err, token) => {
             res.json({
               success: true,
-              token: "Bearer " + token
+              token: "Bearer " + token,
             });
-          });
-        } else {
-          errors.password = "Incorrect password";
-          return res.status(400).json(errors);
-        }
-      });
+          }
+        );
+      } else {
+        errors.password = "Incorrect password";
+        return res.status(400).json(errors);
+      }
     });
   });
+});
 
-//   router.get('/current', passport.authenticate('jwt', {session: false}), (req, res) => {
-//     res.json({msg: 'Success'});
-//   })
-
-  router.get('/current', passport.authenticate('jwt', {session: false}), (req, res) => {
+router.get(
+  "/current",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
     res.json({
       id: req.user.id,
       name: req.user.name,
-      email: req.user.email
+      email: req.user.email,
+      eventsPlanned: req.body.eventsPlanned,
+      eventsHosted: req.body.eventsHosted,
+      friends: req.body.friends,
     });
-  })
+  }
+);
+
+//using req.user.id to ensure we are finding the user from the jwt token and even if the user change the id in the route, they won't be able to update info for someone else
+
+router.patch(
+  "/:id",
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    const { errors, isValid } = validateUserUpdate(req.body);
+
+    if (!isValid) {
+      return res.status(400).json(errors);
+    }
+
+    User.findOneAndUpdate(
+      { _id: req.user.id },
```

Comment:
why is this req.user.id and not req.params.id?

Reply:
because req.user is the authorized logged in user, retrieved from the jwt token